### PR TITLE
Allow JS.patch and JS.navigate to be tested with render_click

### PR DESCRIPTION
**Pull request by nmk**
Fixes https://github.com/phoenixframework/phoenix_live_view/issues/2768

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

**Comment by nmk** [comment]
What is the desired behaviour for something like
```elixir
<button phx-click={JS.patch("/foo") |> JS.navigate("/bar")}>...</button>
```
In this PR the last "navigation command" in the phx-click wins, i.e. it would be a navigate in this case. This only affects the elixir test client though.

IMO the above example should ideally emit a warning.

**Comment by nmk** [comment]
I would love to have that merged.

**Comment by paulstatezny** [comment]
Just ran into this – I hope we can publish this soon!

**Comment by josevalim** [comment]
Folks, saying you would love to see it merged just adds noise and more work on the maintainers and everyone's inbox. If you don't have much to add, you can always use thumbs up on the issue/PR.

**Comment by josevalim** [comment]
> In this PR the last "navigation command" in the phx-click wins, i.e. it would be a navigate in this case. This only affects the elixir test client though.

In the browser, wouldn't the first one win?

File: lib/phoenix_live_view/test/client_proxy.ex [hunk 1]
```diff
@@ -969,8 +969,10 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
     end
   end
 
-  defp maybe_event(:click, {"a", _, _} = node, element) do
-    live_nav = DOM.attribute(node, "data-phx-link")
+  defp maybe_event(:click, {tag, _, _} = node, element) do
+    live_nav =
+      DOM.attribute(node, "data-phx-link") ||
```
Comment (josevalim) [inline lib/phoenix_live_view/test/client_proxy.ex:974]: In the browser, wouldn't phx-click win over data-phx-link?

**Comment by nmk** [comment]
> In the browser, wouldn't the first one win?

I just tried this out in a browser:

`JS.patch(x) |> JS.navigate(x)` acts like `JS.navigate(x)`

`JS.navigate(x) |> JS.patch(x)` is surprising. It reloads the page but does not patch or navigate to the passed in path/url. I don't see any JS errors in the browser console.

You can reproduce the second scenario with:
```elixir
Application.put_env(:sample, Example.Endpoint,
  http: [ip: {127, 0, 0, 1}, port: 5001],
  server: true,
  live_view: [signing_salt: "aaaaaaaa"],
  secret_key_base: String.duplicate("a", 64)
)

Mix.install([
  {:plug_cowboy, "~> 2.5"},
  {:jason, "~> 1.0"},
  {:phoenix, "~> 1.7.10"},
  {:phoenix_live_view, "~> 0.20.1"}
])

defmodule Example.ErrorView do
  def render(template, _), do: Phoenix.Controller.status_message_from_template(template)
end

defmodule Example.HomeLive do
  use Phoenix.LiveView, layout: {__MODULE__, :live}

  def mount(params, _, socket) do
    {:ok, assign(socket, p: Map.get(params, "p", "notset"))}
  end

  def handle_params(params, _, socket) do
    {:noreply, assign(socket, p: Map.get(params, "p", "notset"))}
  end

  defp phx_vsn, do: Application.spec(:phoenix, :vsn)
  defp lv_vsn, do: Application.spec(:phoenix_live_view, :vsn)

  def render("live.html", assigns) do
    ~H"""
    <script src={"https://cdn.jsdelivr.net/npm/phoenix@#{phx_vsn()}/priv/static/phoenix.min.js"}></script>
    <script src={"https://cdn.jsdelivr.net/npm/phoenix_live_view@#{lv_vsn()}/priv/static/phoenix_live_view.min.js"}></script>
    <script>
      let liveSocket = new window.LiveView.LiveSocket("/live", window.Phoenix.Socket)
      liveSocket.connect()
    </script>
    <style>
      * { font-size: 1.1em; }
    </style>
    <%= @inner_content %>
    """
  end

  def render(assigns) do
    ~H"""
    <p><%= @p %></p>

    <button
      type="button"
      phx-click={Phoenix.LiveView.JS.navigate("/nav-test/foo") |> Phoenix.LiveView.JS.patch("/nav-test/foo")}
    >
      Go!
    </button>
    """
  end

end

defmodule Example.Router do
  use Phoenix.Router
  import Phoenix.LiveView.Router

  pipeline :browser do
    plug(:accepts, ["html"])
  end

  scope "/", Example do
    pipe_through(:browser)

    live("/nav-test/:p", HomeLive)
  end
end

defmodule Example.Endpoint do
  use Phoenix.Endpoint, otp_app: :sample
  socket("/live", Phoenix.LiveView.Socket)
  plug(Example.Router)
end

{:ok, _} = Supervisor.start_link([Example.Endpoint], strategy: :one_for_one)
Process.sleep(:infinity)
```

**Comment by josevalim** [comment]
Maybe it tries to patch without a LiveView and that crashes? What happens when we have two patches? Are both applied?

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

File: lib/phoenix_live_view/test/client_proxy.ex (hunk 1 above)
Comment (josevalim) [reply lib/phoenix_live_view/test/client_proxy.ex:974]: Just checking about which one should win here. I think phx-click but I am not sure.

**Pull request merged**